components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/impl/ApisApiServiceImpl.java

```diff
@@ -60,7 +66,30 @@ public Response apisGet(String limit,String offset,String query,String type,Stri
     @Override
     public Response apisPost(API body,String contentType)
     throws NotFoundException {
-        // do some magic!
+
+        boolean isTenantFlowStarted = false;
+        try {
+            org.wso2.carbon.apimgt.api.model.API apiToAdd = MappingUtil.fromDTOtoAPI(body);
+             String tenantDomain =  CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+             if(tenantDomain != null && !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
```

Comment:
Please note that if the tenant domain becomes null, !MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain) becomes true. Therefore it will start a tenant flow for null.
In order to guard this, tenantDomain != null is used.